Lib/subprocess.py

```diff
@@ -1307,15 +1307,15 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
+                    err_msg = err_msg.decode(errors="surrogatepass")
                 except ValueError:
                     exception_name = b'SubprocessError'
                     hex_errno = b'0'
-                    err_msg = (b'Bad exception data from child: ' +
+                    err_msg = ('Bad exception data from child: ' +
                                repr(errpipe_data))
```

Comment:
Sorry if I'm missing something but I don't see how this is functionally any different from having an else block.

With this code the sequence looks like:

1. Try to get the err_msg str by splitting and then decoding the errpipe_data
    - If that fails, err_msg str is errpipe_data decoded and wrapped with 'Bad exception data...'

with the else clause it'll be:

1. Try to get the err_msg bytes by splitting errpipe_data
    - If that fails, get the err_msg bytes by wrapping errpipe_data with b'Bad exception data...' 
2. Decode err_msg into a str